### ISSUE TITLE
Example script for image reward scoring using UniPercept

### ISSUE THREAD

**Issue by YujieOuO** [thunderbolt215/UniPercept#2]
Nice work for image quality assesment! I've created a simple example script that uses UniPercept as a reward model to score images on dimensions like aesthetics, quality, or texture richness. 

And It would be more convenient if the model could be packaged.
```
import os
import sys
import argparse
import torch
from PIL import Image
import torchvision.transforms as T
from transformers import AutoTokenizer
from torchvision.transforms.functional import InterpolationMode

# Add src to path
src_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), "src")
sys.path.insert(0, src_dir)

from src.internvl.model.internvl_chat.modeling_unipercept import InternVLChatModel

IMAGENET_MEAN = (0.485, 0.456, 0.406)
IMAGENET_STD = (0.229, 0.224, 0.225)

def build_transform(input_size=448):
    return T.Compose([
        T.Lambda(lambda img: img.convert("RGB") if img.mode != "RGB" else img),
        T.Resize((input_size, input_size), interpolation=InterpolationMode.BICUBIC),
        T.ToTensor(),
        T.Normalize(mean=IMAGENET_MEAN, std=IMAGENET_STD),
    ])

def load_image(image_file, input_size=448):
    image = Image.open(image_file).convert("RGB")
    transform = build_transform(input_size)
    return transform(image).unsqueeze(0)

def main():
    parser = argparse.ArgumentParser(description="Score an image using UniPercept Reward Model")
    parser.add_argument("--model_path", type=str, required=True, help="Path to the UniPercept model")
    parser.add_argument("--image_path", type=str, required=True, help="Path to the input image")
    parser.add_argument("--desc", type=str, default="aesthetics",
                        choices=["aesthetics", "quality", "structure and texture richness"],
                        help="Scoring dimension")
    args = parser.parse_args()

    # Validate paths
    if not os.path.exists(args.model_path):
        raise FileNotFoundError(f"Model path not found: {args.model_path}")
    if not os.path.isfile(args.image_path):
        raise FileNotFoundError(f"Image file not found: {args.image_path}")

    device = "cuda" if torch.cuda.is_available() else "cpu"
    print(f"Using device: {device}")

    # Load model
    model = InternVLChatModel.from_pretrained(
        args.model_path,
        torch_dtype=torch.bfloat16,
        low_cpu_mem_usage=True,
        use_flash_attn=True,
    )

    model.eval().to(device)

    tokenizer = AutoTokenizer.from_pretrained(args.model_path, trust_remote_code=True, use_fast=False)
    gen_cfg = dict(max_new_tokens=512, do_sample=False)

    pixel_values = load_image(args.image_path).to(torch.bfloat16).to(device)

    with torch.no_grad():
        score = model.score(device, tokenizer, pixel_values, gen_cfg, args.desc)

    print(f"Score ({args.desc}): {score}")

if __name__ == "__main__":
    main()
```

**Comment by thunderbolt215** [comment]
> could the author provide source code for training to reproduce your results of "Baseline(FLUX.1-dev)" and "w / UniPercept All Rewards"? [@thunderbolt215](https://github.com/thunderbolt215)
> 
> also, could UniPercept could handle text-alignment dimension?

@MengHao666 Thanks for your interest!

1. Training for UniPercept as Reward: We use [Flow-GRPO](https://github.com/yifan123/flow_grpo) as our codebase. The training for each reward took about 1 day on 16 $\times$ A100 GPUs. Specifically for UniPercept Reward (All), we simply sum the three rewards and calculate the advantage based on the aggregated score. 
2. Text-Alignment: Since UniPercept takes only images as input, it does not handle text-alignment. For practical application scenarios, we recommend incorporating a specific text-alignment reward alongside UniPercept during post-training.

**Comment by taey16** [comment]
Thanks a lot for kindly sharing of your work. It is not easy with the 'pip' because I need a fully differentiable reward model. The example script gives me a chance to apply 'UniPercept' into my work.
